WordPress/src/main/java/org/wordpress/android/ui/notifications/NotificationsListViewModel.kt

```diff
@@ -27,30 +30,26 @@ import javax.inject.Named
 class NotificationsListViewModel @Inject constructor(
     @Named(UI_THREAD) mainDispatcher: CoroutineDispatcher,
```

Comment:
We could add a background dispatcher and use it in the "like coroutine" to avoid blocking the UI when updating the like status.
`@Named(BG_THREAD) val bgDispatcher: CoroutineDispatcher`

Reply:
I'm gonna replace `@Named(UI_THREAD) mainDispatcher: CoroutineDispatcher` with `@Named(BG_THREAD) val bgDispatcher: CoroutineDispatcher` to see if the blocking issue in main thread can be resolved.

But I guess the root cause is not the about the dispatcher, it should throw `NetworkOnMainThreadException` if I run on the wrong thread.

I noticed that UI refreshes many times because of the event bus, I'd like to remove redundant events after #20024 is done.

Reply:
Makes sense @jarvislin 👍 

> I noticed that UI refreshes many times because of the event bus, I'd like to remove redundant events after https://github.com/wordpress-mobile/WordPress-Android/issues/20024 is done.

Sounds good. We can merge this PR and tackle this later.

Reply:
Hmmm the main thread is blocking even I don't click on the action icon. So there should be a different root cause.